### ISSUE TITLE
Default filebeat.yml should include output.elasticsearch.index

### ISSUE THREAD

**Issue by cawoodm** [elastic/beats#6495]
It's always mystified me why the default filebeat.yml includes so many commented out options most people will never need but the one option everyone needs is not there at all: output.elasticsearch.index

```yml
output.elasticsearch:
  hosts: ["localhost:9200"]
  index: "myindex"
```

Must be an oversight but it's a real stumbling block when starting with FileBeats.

**Comment by cawoodm** [comment]
And why, oh why, is `enabled: false` under `filebeat.prospectors:`? Are we trying to make FileBeats do nothing OOTB?

**Comment by ruflin** [comment]
The reason the index option is not in the short config is because if you change it quite a few things which work out of the box like the template or dashboard stop working. Should the short config have even less options inside: Probably yes.

For the prospectors we made this change with 6.0 to do nothing out of the box but show a good error (hopefully) on what should be done. Previously we shipped by default logs from `/var/log/*.log` which is pretty cool to get started and a demo. But it had the effect that some people shipped logs to a central server which they actually never intended to ship. So instead of having a potentially unexpected default we now require the user to either enable a module with one command or change one line in the config.

**Comment by cawoodm** [comment]
Thanks for the explanation. I must say Logstash is very difficult to get working and I somehow thought Filebeat is supposed to be simpler and easier and it's things like the above which get newcomers stuck and, I think, unnecessarily so.

The basic premise of filebeat is to load logfile entries into an index in ElasticSearch. Sounds like I need to tell FileBeat 3 things: a) which logs, b) which ES server and c) which index.

But this is only possible by adding a line which isn't there and changing an obscure default. Granted, just adding the line doesn't work, and I'm still battling with that but then I have to ask: why does it need to be so complicated? Why do I need templates if I just want to load my logs into a fixed index named "logs"? Maybe I don't want logs-%yyymmdd% or filebeat-%yyymmdd%.

Also, I got no error when enabled was false. Debug logging did not help. I spent 30min googling "filebeat does nothing" before I realised. FileBeat should warn me if it's running but has no prospectors.

I also don't see why Filebeat is so intimately tied to Kibana. The dashboards sound more like a demo than a core filebeat setting yet they take up so much of the yml where critical stuff (above) is missing.

In short I would expect the following yml lines:
```
# 1. Enable log file collection here:
# filebeat.prospectors:
# - type: log
#  enabled: true
#  paths:
#    - /var/whateva/*.log

# 2. Specify your ES here
output.elasticsearch:
#  hosts: ["localhost:9200"]

# 3. Specify your index here
#index: "filebeat-%yyyymmdd%"
```

 At the risk of hijacking my own thread: where do I find a guide on how to connect filebeat to a fixed (non timestamp based) index like "logs" (as opposed to logs-%yyyymmdd%)?

**Comment by ruflin** [comment]
A few notes here:

* Templates and dashboards have become an integral part of Filebeat with the introduction of filebeat modules which create structured data out of log lines. They are based on the powerful prospectors but are a very easy way to get started.
* Changing the index pattern has become harder in 6.0 because it has many more potential traps / side effects. To not fall into these traps too easy, we removed it. But we are definitively working on making this more user friendly again. Better error messages is part of this and some improvements were already made in 6.1 and 6.2 but we are not there yet.
* Filebeat does nothing and no error: This is actually a bug: https://github.com/elastic/beats/issues/6460

**Comment by kvch** [comment]
This issue is for an old version of beats, I suggest to try out the latest version of beats. If this is still a problem in the newly released version feel free to reopen this issue. Thanks.
